crates/llm_gateway/src/stream_context.rs

```diff
@@ -344,6 +420,75 @@ impl HttpContext for StreamContext {
                     }
                 };
             self.response_tokens += token_count;
+
+            // Compute TTFT if not already recorded
+            if !self.ttft_recorded {
+                if let Some(start_time) = self.start_time {
+                    match get_current_time() {
+                        Ok(current_time) => match current_time.duration_since(start_time) {
+                            Ok(duration) => {
+                                let duration_ms = duration.as_millis();
+                                debug!("Time to First Token (TTFT): {} milliseconds", duration_ms);
+                                self.ttft_duration = Some(duration);
+                                self.metrics.time_to_first_token.record(duration_ms as u64);
+                            }
+                            Err(e) => {
+                                warn!("SystemTime error: {:?}", e);
+                            }
+                        },
+                        Err(e) => {
+                            warn!("Failed to get current time: {:?}", e);
+                        }
+                    }
+                    self.ttft_recorded = true;
+                } else {
+                    warn!("Start time was not recorded");
+                }
+            }
+            // Check if first token was not processed yet, and if there are tokens in the response.
+            // If so, set the last_token_time to now and set first_token_processed to true
+            if !self.first_token_processed && token_count > 0 {
```

Comment:
don't need to maintain separate var for `self.first_token_processed`, could use optional last_token_time to achieve similar behavior